lib/programmemory.cpp

```diff
@@ -327,11 +327,13 @@ static void programMemoryParseCondition(ProgramMemory& pm, const Token* tok, con
         if (endTok && findExpressionChanged(vartok, tok->next(), endTok, settings))
             return;
         const bool impossible = (tok->str() == "==" && !then) || (tok->str() == "!=" && then);
-        const ValueFlow::Value& v = then ? truevalue : falsevalue;
-        pm.setValue(vartok, impossible ? asImpossible(v) : v);
+        ValueFlow::Value& v = then ? truevalue : falsevalue;
```

Comment:
Should be `ValueFlow::Value&& v = then ? std::move(truevalue) : std::move(falsevalue)`.